middlewares/accesslog/logger_formatters.go

```diff
@@ -0,0 +1,69 @@
+package accesslog
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// default format for time presentation
+const commonLogTimeFormat = "02/Jan/2006:15:04:05 -0700"
+
+// CommonLogFormatter provides formatting in the Traefik common log format
+type CommonLogFormatter struct {
+}
+
+//Format formats the log entry in the Traefik common log format
+func (f *CommonLogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	b := &bytes.Buffer{}
+
+	timestamp := entry.Data[StartUTC].(time.Time).Format(commonLogTimeFormat)
+	elapsedMillis := entry.Data[Duration].(time.Duration).Nanoseconds() / 1000000
+	referer := entry.Data["request_Referer"]
+	if referer == nil {
+		referer = ""
+	}
+	userAgent := entry.Data["request_User-Agent"]
+	if userAgent == nil {
+		userAgent = ""
+	}
+
+	_, err := fmt.Fprintf(b, "%s - %s [%s] \"%s %s %s\" %d %d \"%s\" %s %d %s %s %dms\n",
+		entry.Data[ClientHost],
+		entry.Data[ClientUsername],
+		timestamp,
+		entry.Data[RequestMethod],
+		entry.Data[RequestPath],
+		entry.Data[RequestProtocol],
+		entry.Data[OriginStatus],
+		entry.Data[OriginContentSize],
+		referer,
+		quoted(userAgent),
+		entry.Data[RequestCount],
+		quoted(entry.Data[FrontendName]),
+		quoted(entry.Data[BackendURL]),
+		elapsedMillis)
+
+	return b.Bytes(), err
+}
+
+func quoted(v interface{}) string {
+	if v == nil {
+		return "-"
+	}
+
+	switch s := v.(type) {
+	case string:
+		if s == "" {
+			return "-"
+		}
+		return `"` + s + `"`
+
+	case fmt.Stringer:
+		return `"` + s.String() + `"`
+	}
+
+	return "-"
+}
```

Comment:
LGTM I've added it in.  It has meant that the log format is not fully backward compatible.  Referer and User Agent fields are now specified as a hyphen if not present, previously they were just missing.  I believe the new format is actually correct, however.